Library/Homebrew/cmd/update-report.rb

```diff
@@ -238,19 +241,25 @@ def output_update_report
     return if new_repository_version.blank?
 
     puts
-    ohai "Homebrew was updated to version #{new_repository_version}"
-    Settings.write "latesttag", new_repository_version
-    if new_repository_version.split(".").last == "0"
+    ohai "Homebrew was updated to version #{new_repository_version}" unless args.quiet?
+    new_major_version, new_minor_version = (new_repository_version.split(".")[0, 2]).map(&:to_i)
+    if old_repository_version.present?
+      old_major_version, old_minor_version = (old_repository_version.split(".")[0, 2]).map(&:to_i)
+    end
+    if !args.quiet? || old_repository_version.blank? || (new_major_version > old_major_version) \
+        || (new_minor_version > old_minor_version)
       puts <<~EOS
         More detailed release notes are available on the Homebrew Blog:
-          #{Formatter.url("https://brew.sh/blog/#{new_repository_version}")}
-      EOS
-    elsif !args.quiet?
-      puts <<~EOS
-        The changelog can be found at:
-          #{Formatter.url("https://github.com/Homebrew/brew/releases/tag/#{new_repository_version}")}
+          #{Formatter.url("https://brew.sh/blog/#{new_major_version}.#{new_minor_version}.0")}
       EOS
     end
+
+    return if new_minor_version.zero? && args.quiet?
```

Comment:
Why return if `new_minor_version` is zero?

Reply:
The existing revision does not show the changelog message when Homebrew is updated to an x.x.0 version. The new code maintains this logic and adds the functionality to show the blog message and changelog message when Homebrew is updated from, for example, 1.3.8 to 1.4.2 but only the blog message when updating 1.3.8 to 1.4.0. This could be reverted to an either or situation.

Reply:
Making messages more redundant has made this conversation outdated.